addon/components/table-of-contents-plugin/ember-nodes/table-of-contents.ts

```diff
@@ -19,11 +19,38 @@ export default class TableOfContentsComponent extends Component<EmberNodeArgs> {
     return this.args.controller;
   }
 
+  get tableOfContentsRange() {
```

Comment:
seems to me you can replace this lookup with `this.args.getPos() + this.args.node.nodeSize`, did you try that?